// @coreui/react/dist/esm/components/form/CFormFeedback.js
import { __rest, __assign } from '../../node_modules/tslib/tslib.es6.js';
import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from '../../node_modules/classnames/index.js';

var CFormFeedback = forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.component, Component = _c === void 0 ? 'div' : _c, invalid = _a.invalid, tooltip = _a.tooltip, valid = _a.valid, rest = __rest(_a, ["children", "className", "component", "invalid", "tooltip", "valid"]);
    return (React.createElement(Component, __assign({ className: classNames((_b = {},
            _b["invalid-".concat(tooltip ? 'tooltip' : 'feedback')] = invalid,
            _b["valid-".concat(tooltip ? 'tooltip' : 'feedback')] = valid,
            _b), className) }, rest, { ref: ref }), children));
});
CFormFeedback.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.elementType,
    invalid: PropTypes.bool,
    tooltip: PropTypes.bool,
    valid: PropTypes.bool,
};
CFormFeedback.displayName = 'CFormFeedback';

export { CFormFeedback };
