// @coreui/react/dist/cjs/components/pagination/CPaginationItem.js
'use strict';

var tslib_es6 = require('../../node_modules/tslib/tslib.es6.js');
var React = require('react');
var PropTypes = require('prop-types');
var index = require('../../node_modules/classnames/index.js');
var CLink = require('../link/CLink.js');

var CPaginationItem = React.forwardRef(function (_a, ref) {
    var children = _a.children, className = _a.className, component = _a.component, rest = tslib_es6.__rest(_a, ["children", "className", "component"]);
    var Component = component !== null && component !== void 0 ? component : (rest.active ? 'span' : 'a');
    return (React.createElement("li", tslib_es6.__assign({ className: index.default('page-item', {
            active: rest.active,
            disabled: rest.disabled,
        }, className) }, (rest.active && { 'aria-current': 'page' })), Component === 'a' ? (React.createElement(CLink.CLink, tslib_es6.__assign({ className: "page-link", component: Component }, rest, { ref: ref }), children)) : (React.createElement(Component, { className: "page-link", ref: ref }, children))));
});
CPaginationItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.elementType,
};
CPaginationItem.displayName = 'CPaginationItem';

exports.CPaginationItem = CPaginationItem;
