// @coreui/react/dist/esm/components/nav/CNav.js
import { __rest, __assign } from '../../node_modules/tslib/tslib.es6.js';
import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import classNames from '../../node_modules/classnames/index.js';

var CNav = forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.component, Component = _c === void 0 ? 'ul' : _c, layout = _a.layout, variant = _a.variant, rest = __rest(_a, ["children", "className", "component", "layout", "variant"]);
    return (React.createElement(Component, __assign({ className: classNames('nav', (_b = {},
            _b["nav-".concat(layout)] = layout,
            _b["nav-".concat(variant)] = variant,
            _b), className), role: "navigation" }, rest, { ref: ref }), children));
});
CNav.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.elementType,
    layout: PropTypes.oneOf(['fill', 'justified']),
    variant: PropTypes.oneOf(['pills', 'tabs', 'underline', 'underline-border']),
};
CNav.displayName = 'CNav';

export { CNav };
