// @coreui/react/dist/cjs/components/sidebar/CSidebarNav.js
'use strict';

var tslib_es6 = require('../../node_modules/tslib/tslib.es6.js');
var React = require('react');
var PropTypes = require('prop-types');
var index = require('../../node_modules/classnames/index.js');

var CNavContext = React.createContext({});
var CSidebarNav = React.forwardRef(function (_a, ref) {
    var children = _a.children, className = _a.className, rest = tslib_es6.__rest(_a, ["children", "className"]);
    var _b = React.useState(''), visibleGroup = _b[0], setVisibleGroup = _b[1];
    var CNavContextValues = {
        visibleGroup: visibleGroup,
        setVisibleGroup: setVisibleGroup,
    };
    return (React.createElement("ul", tslib_es6.__assign({ className: index.default('sidebar-nav', className), ref: ref }, rest),
        React.createElement(CNavContext.Provider, { value: CNavContextValues }, React.Children.map(children, function (child, index) {
            if (React.isValidElement(child)) {
                return React.cloneElement(child, {
                    key: index,
                    idx: "".concat(index),
                });
            }
            return;
        }))));
});
CSidebarNav.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
};
CSidebarNav.displayName = 'CSidebarNav';

exports.CNavContext = CNavContext;
exports.CSidebarNav = CSidebarNav;
