// @coreui/react/dist/cjs/components/spinner/CSpinner.js
'use strict';

var tslib_es6 = require('../../node_modules/tslib/tslib.es6.js');
var React = require('react');
var PropTypes = require('prop-types');
var index = require('../../node_modules/classnames/index.js');
var props = require('../../props.js');

var CSpinner = React.forwardRef(function (_a, ref) {
    var _b;
    var className = _a.className, color = _a.color, _c = _a.component, Component = _c === void 0 ? 'div' : _c, size = _a.size, _d = _a.variant, variant = _d === void 0 ? 'border' : _d, _e = _a.visuallyHiddenLabel, visuallyHiddenLabel = _e === void 0 ? 'Loading...' : _e, rest = tslib_es6.__rest(_a, ["className", "color", "component", "size", "variant", "visuallyHiddenLabel"]);
    return (React.createElement(Component, tslib_es6.__assign({ className: index.default("spinner-".concat(variant), (_b = {},
            _b["spinner-".concat(variant, "-").concat(size)] = size,
            _b["text-".concat(color)] = color,
            _b), className), role: "status" }, rest, { ref: ref }),
        React.createElement("span", { className: "visually-hidden" }, visuallyHiddenLabel)));
});
CSpinner.propTypes = {
    className: PropTypes.string,
    color: props.colorPropType,
    component: PropTypes.string,
    size: PropTypes.oneOf(['sm']),
    variant: PropTypes.oneOf(['border', 'grow']),
    visuallyHiddenLabel: PropTypes.string,
};
CSpinner.displayName = 'CSpinner';

exports.CSpinner = CSpinner;
