// @coreui/react/dist/cjs/components/link/CLink.js
'use strict';

var tslib_es6 = require('../../node_modules/tslib/tslib.es6.js');
var React = require('react');
var PropTypes = require('prop-types');
var index = require('../../node_modules/classnames/index.js');

var CLink = React.forwardRef(function (_a, ref) {
    var children = _a.children, active = _a.active, className = _a.className, _b = _a.component, Component = _b === void 0 ? 'a' : _b, disabled = _a.disabled, rest = tslib_es6.__rest(_a, ["children", "active", "className", "component", "disabled"]);
    return (React.createElement(Component
    // TODO: remove duplicated classes ex. `active active` in `<CListGroupItem>`
    , tslib_es6.__assign({ 
        // TODO: remove duplicated classes ex. `active active` in `<CListGroupItem>`
        className: index.default(className, { active: active, disabled: disabled }) }, (active && { 'aria-current': 'page' }), (Component === 'a' && disabled && { 'aria-disabled': true, tabIndex: -1 }), ((Component === 'a' || Component === 'button') && {
        onClick: function (event) {
            event.preventDefault;
            !disabled && rest.onClick && rest.onClick(event);
        },
    }), { disabled: disabled }, rest, { ref: ref }), children));
});
CLink.propTypes = {
    active: PropTypes.bool,
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.elementType,
    disabled: PropTypes.bool,
};
CLink.displayName = 'CLink';

exports.CLink = CLink;
