// @coreui/react/dist/cjs/components/nav/CNav.js
'use strict';

var tslib_es6 = require('../../node_modules/tslib/tslib.es6.js');
var React = require('react');
var PropTypes = require('prop-types');
var index = require('../../node_modules/classnames/index.js');

var CNav = React.forwardRef(function (_a, ref) {
    var _b;
    var children = _a.children, className = _a.className, _c = _a.component, Component = _c === void 0 ? 'ul' : _c, layout = _a.layout, variant = _a.variant, rest = tslib_es6.__rest(_a, ["children", "className", "component", "layout", "variant"]);
    return (React.createElement(Component, tslib_es6.__assign({ className: index.default('nav', (_b = {},
            _b["nav-".concat(layout)] = layout,
            _b["nav-".concat(variant)] = variant,
            _b), className), role: "navigation" }, rest, { ref: ref }), children));
});
CNav.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.elementType,
    layout: PropTypes.oneOf(['fill', 'justified']),
    variant: PropTypes.oneOf(['pills', 'tabs', 'underline', 'underline-border']),
};
CNav.displayName = 'CNav';

exports.CNav = CNav;
